### DISCUSSION TITLE
Title Bar Context Menu to match Dark Mode

### DISCUSSION THREAD

**Discussion by AndyHTML2012** [microsoft/terminal#14751] [Ideas]
This is a pretty minor idea, but it would be cool if **the terminal app title bar context menu switched to dark mode when you are using the terminal in dark mode**.

**Here is a photo of what it would look like** (taken from settings app when in dark mode on windows 11)**:**
![Dark_Context_Menu](https://user-images.githubusercontent.com/54720001/215219480-90c01d8f-0310-407f-9a34-7d56a7ca69dc.png)

**Here is what the current one looks like when right clicking on titlebar:**
![Current_Context_Menu](https://user-images.githubusercontent.com/54720001/215219543-ecab5795-b915-450e-9c9c-62dcc0d8c1ca.png)

**Comment by zadjii-msft** [comment]
Huh. I dunno how the Settings app does that. 

I bet it is possible call `TerminalTrySetDarkTheme` on the `GetSystemMenu(_window.get(), FALSE)`. That... might just work?

**Reply by zadjii-msft** [reply]
oh dear
https://github.com/mintty/mintty/issues/983

oh dear
https://learn.microsoft.com/en-us/answers/questions/893697/thrilled-to-have-dwmwa-use-immersive-dark-mode-for

Oh boy, I don't think this is documented, whatever it is. Classic us. I'll start a mail thread monday.

**Reply by AndyHTML2012** [reply]
I know that for windows 11 22h2 most (if not all) UWP and winUI2 apps automatically get the dark mode titlebar context menu automatically when system is set to dark mode.

Somehow the git bash console also has dark context menu for the titlebar and notepad++ (when switched to dark mode)

**Reply by zadjii-msft** [reply]
> Somehow the git bash console also has dark context menu 

INTERESTING. Alas, their code is GPL'd so I can't look at it to figure out how. But I'd assume they're using whatever private API it is.

**Reply by AndyHTML2012** [reply]
I know that towards the end of last year they added the dark title at context menu ability to chromium browsers: [here's a link to what I mean](https://winaero.com/chrome-and-edge-titlebar-context-menus-now-support-the-dark-theme-on-windows/amp/)
